packages/fuels-contract/src/contract.rs

```diff
@@ -439,7 +434,7 @@ pub struct ContractCallHandler<D> {
     pub contract_call: ContractCall,
     pub tx_parameters: TxParameters,
     pub wallet: LocalWallet,
-    pub fuel_client: FuelClient,
+    pub provider: Provider,
```

Comment:
All of this is sooo much cleaner, I love it. It had been bugging me for a while lol